include/rocksdb/comparator.h

```diff
@@ -74,6 +74,12 @@ class Comparator {
                                               const Slice& /*t*/) const {
     return false;
   }
+
+  // return true if two keys with different byte sequences can be regarded
+  // as equal by this comparator.
+  // The major use case is to determine if DataBlockHashIndex is compatible
+  // with the customized comparator.
+  virtual bool CanKeysWithDifferentByteContentsEqual() const { return true; }
```

Comment:
Nit: (as this is in the public API):
Add a 'Be' in between. 
s/CanKeysWithDifferentByteContentsEqual/CanKeysWithDifferentByteContents**Be**Equal/